calculator_test.go

```diff
@@ -2,23 +2,304 @@ package calculator_test
 
 import (
 	"calculator"
+	"math/rand"
 	"testing"
 )
 
 func TestAdd(t *testing.T) {
-	t.Parallel()
-	var want float64 = 4
-	got := calculator.Add(2, 2)
-	if want != got {
-		t.Errorf("want %f, got %f", want, got)
+
+	t.Run("Fixed number addition", func(t *testing.T) {
+		var want float64 = 4
+		got := calculator.Add(2, 2)
+		if want != got {
+			t.Errorf("want %f, got %f", want, got)
+		}
+	})
+
+	t.Run("Sum four numbers", func(t *testing.T) {
+		var want float64 = 40
+		got := calculator.Add(2, 8, 20, 10)
+		if want != got {
+			t.Errorf("want %f, got %f", want, got)
+		}
+	})
+
+}
+
+func TestAddRandom(t *testing.T) {
+	var totalRandom int = 100
+	t.Run("Random number addition", func(t *testing.T) {
+		for count := 0; count < totalRandom; count++ {
+			var a float64 = rand.NormFloat64()*1024 + 2
+			var b float64 = rand.NormFloat64()*2048 + 5
+			want := a + b
+			got := calculator.Add(a, b)
+			if want != got {
+				t.Errorf("want %f, got %f", want, got)
+			}
+		}
+	})
+}
+
+func TestSubtract(t *testing.T) {
+	testCases := []struct {
+		name  string
+		want  float64
+		n1    float64
+		n2    float64
+		extra []float64
+	}{
+		{
+			name:  "Substract a > b returns positive",
+			n1:    3,
+			n2:    2,
+			extra: nil,
+			want:  1,
+		},
+		{
+			name:  "Substract a < b returns negative",
+			n1:    3,
+			n2:    5,
+			extra: nil,
+			want:  -2,
+		},
+		{
+			name:  "Ensure decimals",
+			n1:    6,
+			n2:    2.225,
+			extra: nil,
+			want:  3.77500,
+		},
+		{
+			name:  "Substract a and b negatives which returns a positive",
+			n1:    -3,
+			n2:    -5,
+			extra: nil,
+			want:  2,
+		},
+		{
+			name:  "Substract five numbers",
+			n1:    20,
+			n2:    5,
+			extra: []float64{4, 3, 7},
+			want:  1,
+		},
+	}
+	for _, tC := range testCases {
+		t.Run(tC.name, func(t *testing.T) {
+			got := calculator.Subtract(tC.n1, tC.n2, tC.extra)
+			if tC.want != got {
+				t.Errorf("want %f, got %f", tC.want, got)
+			}
+		})
+	}
+}
+
+func TestMultiply(t *testing.T) {
+	testCases := []struct {
+		name  string
+		want  float64
+		n1    float64
+		n2    float64
+		extra []float64
+	}{
+		{
+			name:  "Multiply two positive numbers returns positive",
+			n1:    3,
+			n2:    2,
+			extra: nil,
```

Comment:
Since the default value for slice fields is `nil`, this assignment has no effect; we can just omit the `extra` field when we're not setting it.